java/algorithm/pom.xml

```diff
@@ -48,5 +52,87 @@
     </dependency>
   </dependencies>
 
-  <build></build>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <version>3.1.0</version>
+        <configuration>
+          <skip>true</skip>
```

Comment:
I see your point. Any suggestions, what we can do better here?

Reply:
We may still want to run checkstyle if it has other useful lints, we just wouldn't use checkstyle for code formatting since it's awful at that job.

Reply:
@laurentgo @lidavidm I observed the following when I re-enabled check-style for the algorithm module;

1. EmptyLineSeparator (whitespace)
2. ImportOrder (imports)
3. OperatorWrap (whitespace)
4. RightCurly (blocks)
5. SingleLineJavadoc (javadoc)
6. Indentation (indentation)

I observed the above warnings and ultimately the build fails. I think as both of you suggested we should list checkstyle features we want to keep. But it seems these are in collision course with what spotless is doing. 

I can suggest the following; 

Create a new issue for the graceful update of checkstyle features while working per module. But while working on spotless addition per module, we can disable checkstyle for the moment, but in that separate follow up issue, we remove the colliding checkstyle features and remove checkstyle-disable from each module. 

Appreciate your thoughts.

Reply:
What about creating a build property in parent pom `checkstyle.config.location` defaulting to `dev/checkstyle/checkstyle.xml` and in `algorithm` module changing it to `dev/checkstyle/checkstyle-spotless.xml`?

The new file could be a copy of the original minus the conflicting checkers. I would suggest for the initial revision to comment the checkers instead of removing them to make it easier to review and confirm those are indeed conflicting checkers vs just a config change.

Reply:
Yes, that is a good option. I can try that.

Reply:
@laurentgo I updated accordingly.